openfl/interface/interactive_api/experiment.py

```diff
@@ -0,0 +1,335 @@
+        # Save the prepared plan
+        Plan.Dump(Path('./plan/plan.yaml'), self.plan.config, freeze=False)
+
+        # PACK the WORKSPACE!
+        # Prepare requirements file to restore python env
+        self._export_python_env()
+
+        # Compress te workspace to restore it on collaborator
+        self._pack_the_workspace()
+
+        # DO CERTIFICATES exchange
+
+    def start_experiment(self, model_provider):
+        '''
+        This method also requires model_provider to start experiment with another initialization
+        without workspace redistribution.
+        '''
+        # Start the aggregator
+        self.plan.resolve()
+
+        initial_tensor_dict = self._get_initial_tensor_dict(model_provider)
+        self.server = self.plan.interactive_api_get_server(
+            initial_tensor_dict,
+            chain=self.federation.cert_chain,
+            certificate=self.federation.agg_certificate,
+            private_key=self.federation.agg_private_key)
+
+        self.server.serve()
+        # return server
+
+    @staticmethod
+    def _export_python_env():
+        from pip._internal.operations import freeze
+        requirements_generator = freeze.freeze()
+        with open('./requirements.txt', 'w') as f:
+            for package in requirements_generator:
+                if 'openfl' not in package:
+                    f.write(package + '\n')
+
+    @staticmethod
+    def _pack_the_workspace():
+        from shutil import make_archive, copytree, ignore_patterns, rmtree
+        from os import getcwd, makedirs
+        from os.path import basename
+
+        archiveType = 'zip'
+        archiveName = basename(getcwd())
+        # archiveFileName = archiveName + '.' + archiveType
+
+        tmpDir = 'temp_' + archiveName
+        makedirs(tmpDir)
+
+        ignore = ignore_patterns(
+            '__pycache__', 'data', 'cert', tmpDir, '*.crt', '*.key',
+            '*.csr', '*.srl', '*.pem', '*.pbuf')
+
+        copytree('./', tmpDir + '/workspace', ignore=ignore)
+
+        make_archive(archiveName, archiveType, tmpDir + '/workspace')
+
+        rmtree(tmpDir)
+
+    def _get_initial_tensor_dict(self, model_provider):
+        self.task_runner_stub = self.plan.get_core_task_runner(model_provider=model_provider)
+        tensor_dict, _ = split_tensor_dict_for_holdouts(
+            self.logger,
+            self.task_runner_stub.get_tensor_dict(False),
+            **self.task_runner_stub.tensor_dict_split_fn_kwargs
+        )
+        return tensor_dict
+
+    def _prepare_plan(self, model_provider, task_keeper, data_loader,
+                      rounds_to_train,
+                      delta_updates=False, opt_treatment='RESET',
+                      model_interface_file='model_obj.pkl', tasks_interface_file='tasks_obj.pkl',
+                      dataloader_interface_file='loader_obj.pkl'):
+        # Load the default plan
+        with resources.path('openfl.interface.interactive_api', 'plan.yaml') as plan_path:
```

Comment:
`plan.yaml` in the interactive_api folder does not get copied to this location during openfl installation. It would be good to either move this file to an artifact directory and rework `setup.py` so this gets copied properly, or have it reside in a default workspace directory under `openfl-workspace`. The advantage of the second approach is openfl-workspace continues to be the central location for all plan templates

Reply:
Moved default plan for the Interactive python API to `openfl/openfl-workspace/workspace/plan/plans/default/base_plan_interactive_api.yaml`